src/Http/Http.Results/test/ServerSentEventsResultTests.cs

```diff
@@ -0,0 +1,241 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Net.ServerSentEvents;
+using System.Reflection;
+using System.Text;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http.Json;
+using Microsoft.AspNetCore.Http.Metadata;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.AspNetCore.Routing.Patterns;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.AspNetCore.Http.HttpResults;
+
+public class ServerSentEventsResultTests
+{
+    [Fact]
+    public async Task ExecuteAsync_SetsContentType()
+    {
+        // Arrange
+        var httpContext = GetHttpContext();
+        var events = AsyncEnumerable.Empty<string>();
+        var result = TypedResults.ServerSentEvents(events);
+
+        // Act
+        await result.ExecuteAsync(httpContext);
+
+        // Assert
+        Assert.Equal("text/event-stream", httpContext.Response.ContentType);
+    }
+
+    [Fact]
+    public async Task ExecuteAsync_WritesStringEventsToResponse()
+    {
+        // Arrange
+        var httpContext = GetHttpContext();
+        var events = new[] { "event1\"with\"quotes", "event2" }.ToAsyncEnumerable();
+        var result = TypedResults.ServerSentEvents(events);
+
+        // Act
+        await result.ExecuteAsync(httpContext);
+
+        // Assert
+        var responseBody = Encoding.UTF8.GetString(((MemoryStream)httpContext.Response.Body).ToArray());
+        Assert.Contains("data: event1\"with\"quotes\n\n", responseBody);
+        Assert.Contains("data: event2\n\n", responseBody);
+
+        // Verify strings are not JSON serialized
+        Assert.DoesNotContain("data: \"event1", responseBody);
+        Assert.DoesNotContain("data: \"event2", responseBody);
+    }
+
+    [Fact]
+    public async Task ExecuteAsync_WritesStringsEventsWithType()
+    {
+        // Arrange
+        var httpContext = GetHttpContext();
+        var events = new[] { "event1" }.ToAsyncEnumerable();
```

Comment:
nit: have multiple items and verify event is written multiple times

Reply:
Done.